### PR TITLE
Build fixes for the sky_engine and sky_services Dart packages

### PR DESCRIPTION
**Pull request by jason-simmons**
* Add some missing dependecies to sky_services
* Create a script that excludes unnecessary files when preparing the package for release

### CODE REVIEW

File: sky/tools/dist_dart_pkg.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  parser.add_argument('--dest', type=str, help='Destination directory for the package')
+
+  args = parser.parse_args()
+
+  if os.path.exists(args.dest):
+    shutil.rmtree(args.dest)
+
+  # dart_pkg.py will create a packages directory within the package.
+  # Do not copy this into the release output.
+  shutil.copytree(args.source, args.dest, ignore=shutil.ignore_patterns('packages'))
```
Comment (abarth) [inline sky/tools/dist_dart_pkg.py]: Will this resolve symlinks?

File: sky/dist/BUILD.gn [hunk 2]
```diff
@@ -64,32 +64,47 @@ copy("sky_shell") {
...
+
     deps = [
       "//sky/packages/sky_engine",
     ]
   }
 
-  copy_ex("sky_services") {
-    clear_dir = true
-    # Note: The package actually ends up in $root_dist_dir/packages/sky_services/sky_services
+  action("sky_services") {
```
Comment (abarth) [inline sky/dist/BUILD.gn]: If you run "ninja :dist" repeatedly, will this action run repeatedly?  That's usually undesirable but in this case is probably fine.

File: sky/tools/dist_dart_pkg.py (hunk 1 above)
Comment (jason-simmons) [inline sky/tools/dist_dart_pkg.py]: yes

File: sky/dist/BUILD.gn (hunk 2 above)
Comment (jason-simmons) [inline sky/dist/BUILD.gn]: The action will run only if there's a change to the package listed in the dependencies

Comment (abarth) [inline sky/dist/BUILD.gn]: Ok, so we need to be careful about the dependencies of the `//sky/packages/sky_services` until we make that more robust.

**Comment by abarth** [comment]
LGTM

**Pull request merged**